### PR TITLE
[#271] Restricted variable names with more than one consecutive underscore

### LINKED ISSUE #271: Restrict variable names with more than one consecutive underscore
# Rule request

## Thesis

Currently it is possible to name your variables like: `some__value`. This should be forbidden.

Mind, that `__magic__` variables can still be used.

## Reasoning

This is done to gain extra readability. This naming rule already exist for module names.

### PR DESCRIPTION
**Pull request by Roxe322**
# I have made things!

## Checklist

- [V] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [V] I have created at least one test case for the changes I have made
- [V] I have updated the documentation for the changes I have made
- [V] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #271

### CODE REVIEW

**Comment by Roxe322** [comment]
@sobolevn I'm really not sure that I've made some right things. First is moving `logics/variable` to module and second is my test (new file and breaking DRY in it). Please, help me with solving it, because I'm really afraid that my decisions in this was wrong

**Comment by Roxe322** [comment]
Also, I've just noticed that `pyproject.lock` was deleted. How can I resolve it? Probably it's because I've installed it with last version of poetry and now I have `poetry.lock` file instead of that.

File: wemake_python_styleguide/logics/variables/name_helpers.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+import ast
```
Comment (sobolevn) [inline wemake_python_styleguide/logics/variables/name_helpers.py]: Missing empty line after `# -*-`

File: wemake_python_styleguide/violations/naming.py [hunk 2]
```diff
@@ -521,3 +524,28 @@ class A(object):
...
+
+    Reasoning:
+        This is done to gain extra readability.
+        This naming rule already exist for module names.
+
+    Example::
+        # Correct:
+        some_value = 5
+        __magic__ = 5
+        # Wrong:
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: There should be an empty line here

File: wemake_python_styleguide/violations/naming.py [hunk 3]
```diff
@@ -521,3 +524,28 @@ class A(object):
...
+class ConsecutiveUnderscoresInNameViolation(ASTViolation):
+    """
+    Forbids to use more than one consecutive underscore in variable names.
+
+    Reasoning:
+        This is done to gain extra readability.
+        This naming rule already exist for module names.
+
+    Example::
+        # Correct:
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: There should be an empty line here before `# Correct:`

File: wemake_python_styleguide/violations/naming.py [hunk 4]
```diff
@@ -521,3 +524,28 @@ class A(object):
...
+        some_value = 5
+        __magic__ = 5
+        # Wrong:
+        some__value = 5
+    Note:
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: There should be an empty line here before `Note:`

**Comment by sobolevn** [comment]
@Roxe322 considering your thoughts about `dry` inside the tests.
Yes, this is a problem. But, we are going to solve it separately.

**Comment by Roxe322** [comment]
Hi, @sobolevn , I've fixed missing blank lines and renamed modules as you suggested. What should I do with DRY problem in tests? Also, I've just noticed that Travis broke in 3.7 tests. Why such problem occured and why in 3.6 everything is ok? I'm using 3.7 on my local machine and pytest working normally.

**Comment by sobolevn** [comment]
You have the same problem as https://github.com/wemake-services/wemake-python-styleguide/pull/280

I am not sure what happens here.

**Comment by vsmaxim** [comment]
Hi, @Roxe322, travis ci is failing due to wrong tests. For more info check this: #283.

**Comment by Roxe322** [comment]
Thank you, @vsmaxim. I'll wait @sobolevn to merge your PR, than pull it and update mine :)

**Comment by Roxe322** [comment]
Thank you, @vsmaxim. I'll wait @sobolevn to merge your PR, than pull it and update mine :)

**Comment by Roxe322** [comment]
Thank you, @vsmaxim. I'll wait @sobolevn to merge your PR, than pull it and update mine :)

**Comment by sobolevn** [comment]
@Roxe322 can you please `.. versionadded:: 0.3.0` to you violation docs? And then I will merge it.

**Comment by Roxe322** [comment]
@sobolevn Ok, also i'll fix merge conflicts

**Comment by sobolevn** [comment]
@Roxe322 please, check your build status.

**Comment by Roxe322** [comment]
@sobolevn Yes, I'm working on this, just not very successful merge conflicts resolving

**Comment by sobolevn** [comment]
Awesome! Thanks!

**Pull request merged**